src/full/Agda/Interaction/Options/Base.hs

```diff
@@ -809,7 +818,10 @@ withKFlag :: Flag PragmaOptions
 withKFlag o = return $ o { optWithoutK = Value False }
 
 withoutKFlag :: Flag PragmaOptions
-withoutKFlag o = return $ o { optWithoutK = Value True }
+withoutKFlag o = return $ o
+  { optWithoutK = Value True
+  , optFlatSplit = setDefault False (optFlatSplit o)
```

Comment:
The changelog only says it is disabled by default with `--cubical-compatible`, but apparently also with `--without-K`.

Maybe update the changelog, if this was intended here.